exercises/Robert_Applin/ex02_oo_basics/inc/ShapeSorter.h

```diff
@@ -0,0 +1,29 @@
+#ifndef SHAPESORTER_H_
+#define SHAPESORTER_H_
+
+#include <ostream>
+#include <vector>
+
+// Forward declare Shape
+class Shape;
+
+class ShapeSorter {
+public:
+  ShapeSorter(std::vector<Shape const *> &&shapes);
```

Comment:
I've been told that it is a raw non-owning pointer and so it is okay to be used. Does that sound right, because I don't know?

Reply:
I added this comment before seeing how you used it further down in the code. In this instance this works well because (as you say) `ShapeSorter` doesn't really own the pointers and you've been careful to make sure the lifetime of the shapes is longer than the `ShapeSorter`. And you are right that it avoid ref counting where (in this case) it's not needed.

I think you are correct to use a raw pointer here. 

I think the general advice still stands. But I will now caveat my advice "with unless you need a non-owning pointer to observe an object". 😄